pyproject.toml

```diff
@@ -51,7 +51,7 @@ gitpython = "^3.1.18"
 packaging = ">=24.1"
 passlib = { extras = ["bcrypt"], version = "~1.7.4" }
 psutil = ">=5.0.0"
-pydantic = "~2.8"
+pydantic = "~2.11"
```

Comment:
can we do `>=2.x, <3`? @stefannica @bcdurak

Reply:
looking at the amount of manual fixes that Baris had to make to allow 2.11 to work, I would say no

Reply:
Most of the changes are coming from linting, however, if you take a look at `models/__init__.py`, we had some changes related to the pydantic upgrade as well. I am not sure why this was not failing before, but if we would relax this 2 weeks ago, the release of pydantic 2.11 would make `zenml` fail completely.

Reply:
@bcdurak can you try this out? it would be a big headache solver

Reply:
I'll try this out

Reply:
I had to remove this `mlserver` requirement from the mlflow integration. It's the only thing that was interfering with us relaxing the Pydantic version range, because it was explicitly installing an older version of Pydantic.

Reply:
But what impact does that have on mlflow examples @stefannica ?

Reply:
@htahir1 absolutely no impact on any of our examples. This `mlserver` feature is turned off by default and none of our examples are setting it explicitly. In fact, if you search `mlserver` in all of the zenml-io GitHub org, you'll not find one mention of it in a demo, example or even notebook.

Reply:
@stefannica hmm wonder why we even have this then. im just a bit concerned about it , maybe it isn't mentioned explicitly but used in the e2e example?

Reply:
`mlserver` is just one flavor of a local mlflow model server. The default one works just as well. I think this feature was never really used, even though it was available.

Reply:
ok then!